tracer/src/Datadog.Trace/AppSec/Waf/ReturnTypes.Managed/InitResult.cs

```diff
@@ -65,14 +64,37 @@ private InitResult(ushort failedToLoadRules, ushort loadedRules, string ruleFile
 
         internal bool Reported { get; set; }
 
-        internal static InitResult FromUnusableRuleFile() => new(0, 0, string.Empty, new Dictionary<string, string[]>(), unusableRuleFile: true);
+        internal static InitResult FromUnusableRuleFile() => new(0, 0, string.Empty, new Dictionary<string, object>(), unusableRuleFile: true);
 
-        internal static InitResult From(DdwafRuleSetInfo ddwaRuleSetInfo, IntPtr? wafHandle, WafLibraryInvoker wafLibraryInvoker)
+        internal static InitResult From(IntPtr diagnostics, IntPtr? wafHandle, WafLibraryInvoker? wafLibraryInvoker)
         {
-            var ddwafObjectStruct = ddwaRuleSetInfo.Errors;
-            var errors = ddwafObjectStruct.Decode();
-            var ruleFileVersion = Marshal.PtrToStringAnsi(ddwaRuleSetInfo.Version);
-            return new(ddwaRuleSetInfo.Failed, ddwaRuleSetInfo.Loaded, ruleFileVersion!, errors, wafHandle: wafHandle, wafLibraryInvoker: wafLibraryInvoker);
+            ushort failedCount = 0;
+            ushort loadedCount = 0;
+            string rulesetVersion = string.Empty;
+            Dictionary<string, object>? errors = null;
+            try
+            {
+                if (diagnostics != IntPtr.Zero)
+                {
+                    var diagnosticsData = (Dictionary<string, object>)Encoder.Decode(new Obj(diagnostics));
+                    if (diagnosticsData.Count > 0)
+                    {
+                        var rules = (Dictionary<string, object>)diagnosticsData["rules"];
+                        failedCount = (ushort)((object[])rules["failed"]).Length;
+                        loadedCount = (ushort)((object[])rules["loaded"]).Length;
+                        errors = (Dictionary<string, object>)rules["errors"];
+                        rulesetVersion = (string)diagnosticsData["ruleset_version"];
+                    }
+                }
+            }
+            catch (Exception err)
+            {
+                var errorMsg = err.ToString();
+                Log.Warning("AppSec could not read Waf diagnostics. Disabling AppSec : {ErrorMsg}", errorMsg);
```

Comment:
here we should Log an error instead of warning I think

Reply:
and we should also add the error to the errors dictionary passed to `InitResult`